### DISCUSSION TITLE
admonitions

### DISCUSSION THREAD

**Discussion by abotzki** [LiaScript/LiaScript#212] [Q&A] [answered]
Hi all,

does anyone have a suggestion how you could create the admonitions which are available in mkdocs in Liascript?
https://squidfunk.github.io/mkdocs-material/reference/admonitions/#admonition-icons-octicons

I thought about a combination of the blockquotes or the code blocks with icons from awesome fonts.

Thanks for your feedback!

**Comment by andre-dietrich** [comment, marked as answer]
Hi @abotzki ... I tried your idea out ... basically this can be done like in the following example. Only some CSS is required, these can then be attached to any block ...

``` markdown
<!--

@style
.admonition {
    padding: 15px 20px;
    margin-bottom: 20px;
    border: 1px solid transparent;
    border-radius: 4px;
    box-shadow: 0 1px 1px rgba(0,0,0,0.05);
    position: relative;
    line-height: 1.4;
}
.admonition::before {
    margin-left: -5rem;
}
.admonition.note {
    color: #31708f;
    background-color: #d9edf7;
    border-left: 5px solid #31708f;
}
.admonition.warning {
    color: #8a6d3b;
    background-color: #fcf8e3;
    border-left: 5px solid #8a6d3b;
}
.admonition.danger {
    color: #a94442;
    background-color: #f2dede;
    border-left: 5px solid #a94442;
}
.admonition-title {
    font-weight: bold;
    margin-bottom: 10px;
}
@end


@admonition: @admonition_(@uid,@0,@1)

@admonition_: <script modify="false">
  setTimeout(() => {
    const parent = document.getElementById('admon-@0').parentElement.parentElement.parentElement.parentElement;

    parent.classList.add("admonition");
    parent.classList.add("@1");

  }, 100);
  `LIASCRIPT: <div class="admonition-title" id="admon-@0">@2</div>`
  </script>

-->

# Admonition

<!-- class="admonition note" -->
> <!-- class="admonition-title" --> This is a test
>
> Foo
> bar
> sadsfasfd

---

> @admonition(warning,This is a test)
>
> Here is a script used, which identifies the parent blockquote
> and attaches the required class names...

---

> @admonition(danger,This is a test)
>
> Same as above ...

---

If nothing works, you can also directly use HTML

<section class="admonition note">
<h5 class="admonition-title">NOTE</h5>
This is a note admonition
</section>
```

You can try this out here in the [LiveEditor](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA5VUW2/bIBh951cw96GOFBzntqauY7mbOqlStU1b3lvsDztoNkRA2kZV//vwLbc11ZoLsuFwDufwQfiJEIRibTYFQx6FUgpuuBT4BWH7WVEALvIAD6erZzzyV89XdX9JVc4FSaQxsgz2BhKpgCmLt3AtCw7YKCr0iiomzD6EKAp8rQM82U19JnpJQT4F2K8Jqr/KE+r6/frr+dNeg11JXS8zwIoV1PBH1vQXXDCyZDxfGrsGb3KFXvddBUHCMqlY6651UbDMoslUsfII7wlpOnAqC2mNnY2HF/4sa5dM0z+5kmsBpBuGSwbZxYHThn+6TWRLcaj1RJWwWR/JzehnGCcn5bI0m7Hx+3IdxaEcUJEzdaRGLyeTyei02ggYsPfVOooDNWK4KbokMykMeWo3KZEFvFlTw7qmXlHMBCBbonu7iPde7t14zaEf+/142DuA3Qc41KniK4NLCTzbzJ2MFpo5kZXTzCx4yeTauG4Pz6NtDkIb3FQrnmOQ6bq0j17OzE3Bqscvm1twz2sVEvvnPa8Bt6P//3aF2gNW9XlpQbW+49rYzMB1diacruDfxMXDatwCXvs2L7/GPtzdXv/++uv258L6B/6I6zlz53gzHMyh7bVGnCgehQMLjx4sRzhogosQIsQ2Z/h6Oxuh0N4Z/7Li6qw4uMJH+G1IJ2wxeLHkGtsfxYZpg6pJ36S0bUKVbTUFnVGdQbUCe0NF+3vutkelf0jS+yBLcwI+TBJqltaGT0QQoXA5PWk++v5jcRMOltMI7YTra2YHRTb/RiP6C/cWrxigBQAA)

Unfortunately, I could not find the CSS file of MkDocs on a CDN, I found this one:

https://www.jsdelivr.com/package/npm/marked-admonition-extension

This can be imported like this, only by linking the CSS file ... The previous commands can be applied similarly. With the additional style I tried to reset the tiny font-size to the one used in LiaScript.

``` markdown
<!--
link: https://cdn.jsdelivr.net/npm/marked-admonition-extension@0.0.4/dist/index.min.css

@style
.admonition {
    font-size: initial !important;
}
@end
-->

# Admonition 2

<div class="admonition admonition-info">
<p class="admonition-title">Publish ESM and CJS in a single package</p>

In the past decade, due to the lack of a standard module system of `JavaScript`, __CommonJS__ (a.k.a the <code>require(&#39;xxx&#39;)</code> and `module.exports` syntax) has been the way how Node.js and NPM packages work. Until 2015, when ECMAScript modules finally show up as the standard solution, the community start migrating to native ESM gradually.
</div>
```

For completeness, here is the link to the [LiveEditor](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA2VSTW/bMAy961ewKTC0QCxn3XZY6hkpgh4WIEWBYOdEtZhYs0x5puw4K/bfJznrBzBdRJB8j49Pyi6SRFhD1RxK7xuep2mhSf5kjdb0rST0KTV1Wqu2Qp0oXTsy3jhKcPBIHKLFTM7k51Qb9qkhjYOsDcmCWYgF+5NFId9g8CwgnL0jn7D5jXMwsaAsXJi6ca1X5G/FH7FA0iJJciEu4e4NfSNEpk0PhVXM3ybveN8pM7R3k1xkzf9tiTfe4iR/7J6s4RLuN2tQpGG52gQhoIANHSxCo4pKHTBLm6DgO4EvY449aCyUxinoDsG7MW9DL7h9BAf1WrUaaqe7wMIn9ljH2m6lerUpWtP43RS226Wrg6TVZruFKyUrqUamrHAa8xZ/dabFqw+Xn77eDsMw3tdZOhZHtbszv8QhOsa7MIi8Gq6hVAxPiGe9R3WC0h3hIeDCg47Ih8f1y24MR9dWEn6QNxZuZh+/TOFYBuz9cn13lvpvD4a9IWXtCTjydQ2EMXHC677sbBftnY7pIizXBb9PsaENLObQKh+MjY5RiHocjQ9Z3UVeKbLwffpc/AVylIoOjgIAAA==)
